File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03338/p03338_num0_parsed.py
# Read input from standard input
n = int(input())
s = input()

# Initialize a set to store the unique letters in the string
unique_letters = set()

# Iterate over the string and add each letter to the set
for letter in s:
    unique_letters.add(letter)

# Initialize a variable to store the maximum number of different letters
max_diff_letters = 0

# Iterate over all possible positions to cut the string
for i in range(1, n):
    # Split the string at position i
    x = set(s[:i])
    y = set(s[i:])
    
    # Compute the number of different letters in both X and Y
    diff_letters = len(x.intersection(y))
    
    # Update the maximum number of different letters
    max_diff_letters = max(max_diff_letters, diff_letters)

# Print the largest possible number of different letters
print(max_diff_letters)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03338/p03338_num0_parsed.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_set>

int main() {
    int N;
    std::cin >> N;
    std::string S;
    std::cin >> S;

    int maxDiffLetters = 0;
    for (int i = 0; i < N - 1; i++) {
        std::unordered_set<char> lettersX;
        std::unordered_set<char> lettersY;

        // Count letters in X
        for (int j = 0; j <= i; j++) {
            lettersX.insert(S[j]);
        }

        // Count letters in Y
        for (int j = i + 1; j < N; j++) {
            lettersY.insert(S[j]);
        }

        // Count different letters in both X and Y
        int diffLetters = 0;
        for (char c : lettersX) {
            if (lettersY.count(c) > 0) {
                diffLetters++;
            }
        }

        maxDiffLetters = std::max(maxDiffLetters, diffLetters);
    }

    std::cout << maxDiffLetters << std::endl;

    return 0;
}